internal/testutil/acc/config_formatter.go

```diff
@@ -50,3 +57,184 @@ func sortStringMapKeys(m map[string]string) []string {
 	sort.Strings(keys)
 	return keys
 }
+func sortStringMapKeysAny(m map[string]any) []string {
+	keys := make([]string, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+func ToSnakeCase(str string) string {
+	snake := matchFirstCap.ReplaceAllString(str, "${1}_${2}")
+	snake = matchAllCap.ReplaceAllString(snake, "${1}_${2}")
+	return strings.ToLower(snake)
+}
+
+func ClusterResourceHcl(projectID string, req *ClusterRequest) (configStr, clusterName string, err error) {
```

Comment:
Main function that uses a few helper functions to avoid a lot of hard-coded field names:
- addPrimitiveAttributesViaJSON for structs
- addPrimitiveAttributes for map[string]any
- addPrimitiveAttributes for hcl/terraform syntax, e.g. `depends_on = [mongodbatlas_project.project_execution]` (notice no quotes )